birdhouse/config/jupyterhub/docker-compose-extra.yml

```diff
@@ -40,6 +40,8 @@ services:
     # the Hub and the rest of the stack are on network 'default'
       - default
       - jupyterhub_network
+    # ensure the deployment does not fail when upgrading Jupyterhub version
+    command: bash -c "jupyterhub upgrade-db && jupyterhub"
```

Comment:
@tlvu 
I would like to have your opinion whether auto upgrade-db seems reasonable? Or do we purposely want this operation to be manual to allow some backups as needed beforehand? Is it possible that this corrupts the database?

Reply:
I've tried it before and it does make a backup before performing the upgrade so if a corruption happens, it's reversible.  Maybe this backup only happens because we used the file based DB, if it had been Postgres, this auto backup won't happen.

I just wonder why the Jupyterhub team did not want auto updating the DB in the first place.

Anyways, for our specific setup, it's probably fine.